src/store/reducers/cart.ts

```diff
@@ -0,0 +1,22 @@
+import {
+    ADD_TO_CART,
+    CartActionTypes,
+    CartState,
+  } from '../../types/cart';
+  
+  const initialState : CartState = {
+    products: [],
```

Comment:
Instead of making `cartReducer` operate on a nested object, this is a great use case for combineReducers, where the cartReducer is directly managing the products array (and also can be even stronger typed as Product[]`). This helps prevent a refactor later, or having the cartReducer expand into a mega reducer as the `CartState` changes.

eg.
```ts
const productsReducer = (state: Product[] = [], action: CartActionTypes): Product[] => {
  // etc
}

const cartReducer = combineReducers({
   products: productsReducer,
})
```

Reply:
Once you're at this point, the cartState is equal to the `ReturnType<typeof cartReducer>`, so the `CartState` can be removed, and instead directly tied to the reducer.

Reply:
done